homeassistant/components/fritzbox/config_flow.py

```diff
@@ -102,7 +110,7 @@ async def async_step_user(self, user_input=None):
             result = await self.hass.async_add_executor_job(self._try_connect)
 
             if result == RESULT_SUCCESS:
-                return self._get_entry()
+                return self._get_entry(self._name)
```

Comment:
Here `self._name` should be just `str`.

Probably you need to add explicit annotation for that:
```python
self._name: str = user_input[CONF_HOST]
```

You can actually define another `TypedDict` for `user_input` which matches `vol.Schema`.

Reply:
with own `TypedDict` for `user_input` again the supertype mismatch occurs :confused: 
```
error: Argument 1 of "async_step_user" is incompatible with supertype "ConfigFlow"; supertype defines the argument type as "Optional[Dict[str, Any]]"
```

Reply:
I see. Let's just use `cast` then.

Reply:
Or use can `cast` for the whole `user_input`:
```python
user_input = cast(UserInput, user_input)
```

Reply:
mhhh ... I'm not sure if his will be enough, since in `async_step_confirm()`  `_get_entry()` is also called, but `self._name` is not set inside it, so still getting `homeassistant/components/fritzbox/config_flow.py:164: error: Argument 1 to "_get_entry" of "FritzboxConfigFlow" has incompatible type "Optional[str]"; expected "str"  [arg-type]`

Reply:
`async_step_confirm` only runs after `async_step_ssdp` which sets `self._name`, so you can add an assert there.

Reply:
easier than thought :see_no_evil: